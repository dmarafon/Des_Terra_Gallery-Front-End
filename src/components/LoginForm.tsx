import { useState } from "react";
import { loginUserThunk } from "../redux/thunks/userThunks";
import { useAppDispatch } from "./hooks";

const LoginForm = (): JSX.Element => {
  const formInitialState = { email: "", password: "" };

  const [formData, setFormData] = useState(formInitialState);

  const dispatch = useAppDispatch();

  const loginSubmit = (event: { preventDefault: () => void }) => {
    event.preventDefault();
    const dispatchedData = { ...formData };
    resetForm();

    dispatch(loginUserThunk(dispatchedData));
  };

  const changeData = (event: { target: { id: any; value: any } }) => {
    setFormData({ ...formData, [event.target.id]: event.target.value });
  };

  const resetForm = () => {
    setFormData(formInitialState);
  };

  return (
    <form onSubmit={loginSubmit} noValidate autoComplete="off">
      <label htmlFor="email">EMAIL</label>
      <input id="email" value={formData.email} onChange={changeData} />
      <label htmlFor="password"> PASSWORD </label>
      <input
        id="password"
        type="password"
        value={formData.password}
        onChange={changeData}
      />
      <button type="submit">SIGN IN</button>
    </form>
  );
};

export default LoginForm;
