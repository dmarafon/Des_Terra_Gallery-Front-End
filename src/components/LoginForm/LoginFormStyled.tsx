import styled from "styled-components";

const LoginFormStyled = styled.div`
  display: flex;
  text-decoration: none;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  flex-direction: row;
  @media (max-width: 920px) {
    flex-direction: column;
  }

  .login__image--container {
    display: flex;
    flex-direction: column;
    @media (max-width: 920px) {
      display: none;
    }
    margin-right: 10vw;
  }

  .login__image--image_container {
    display: flex;
    justify-content: flex-end;
  }
  .login__image {
    max-width: 30vw;
    animation: fadeIn 2s;
    -webkit-animation: fadeIn 1.5s;
    -moz-animation: fadeIn 1.5s;
    -o-animation: fadeIn 1.5s;
    -ms-animation: fadeIn 1.5s;
  }

  .login__image {
  }
  @keyframes fadeIn {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }

  @-moz-keyframes fadeIn {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }

  @-webkit-keyframes fadeIn {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }

  @-o-keyframes fadeIn {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }

  @-ms-keyframes fadeIn {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }

  .login__image--text_container {
    display: flex;
    justify-content: flex-end;
    @media (max-width: 920px) {
      display: none;
    }
  }
  .login__image--subtitle {
    font-size: 0.8vw;
  }

  .login__form--container {
    display: flex;
  }

  .login__container {
    display: flex;
    text-align: center;
    font-family: "Karla", sans-serif;
    text-decoration: none;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-wrap: wrap;
    flex-direction: column;
  }

  .login__text--intro {
    color: #000000;
    font-size: 150%;
    font-weight: 700;
    text-align: center;
    margin-left: 0.006%;
    margin-bottom: 0;
    font-weight: 800;
  }

  .login__text--intro_second {
    color: #000000;
    font-size: 150%;
    font-weight: 700;
    text-align: center;
    margin-left: 0.006%;
    margin: 0;
    font-weight: 800;
  }

  .login__text--register {
    color: #000000;
    font-size: 150%;
    font-weight: 700;
    text-align: center;
    margin-left: 0.006%;
    margin-top: 100px;
    cursor: pointer;
  }

  .login__register--navigation {
    text-decoration: none;
    color: black;
    cursor: pointer;
  }

  .login__text--colored {
    color: #fa803b;
  }

  .login__form {
    font-size: 500%;
    font-weight: 700;
    text-align: center;
    margin-left: 0.006%;
    display: flex;
    padding: 30px;
    @media (min-width: 1218px) {
      width: 500px;
    }
    @media (min-width: 1025px) and (max-width: 1280px) {
      width: 450px;
    }

    @media (min-width: 768px) and (max-width: 1024px) {
      width: 400px;
    }

    @media (min-width: 481px) and (max-width: 767px) {
      width: 350px;
    }
    @media (max-width: 480px) {
      width: 300px;
    }

    height: auto;
    overflow: hidden;
    border-radius: 10px;
    justify-content: center;
    align-items: center;
    flex-wrap: wrap;
    justify-content: flex-start;
  }

  .login__label {
    text-align: left;
    font-size: 15px;
    color: #000000;
    cursor: pointer;
  }

  .login__label,
  .login__input {
    float: left;
    clear: both;
  }

  .login__input {
    font-size: 2.2vh;
    font-family: "Karla", sans-serif;
    margin: 15px 0;
    padding: 15px 10px;
    width: 100%;
    outline: none;
    border: 1px solid #000000;
    border-radius: 5px;
    display: inline-block;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
    -webkit-transition: 0.2s ease all;
    -moz-transition: 0.2s ease all;
    -ms-transition: 0.2s ease all;
    -o-transition: 0.2s ease all;
    transition: 0.2s ease all;
  }

  .login__input[type="text"]:focus,
  .login__input[type="password"]:focus {
    border-color: #000000;
  }

  .login__button--container {
    display: flex;
    align-content: center;
    flex-wrap: wrap;
    flex-direction: column;
  }

  .login__button {
    cursor: pointer;
    justify-content: center;
    height: 50px;
    width: 150px;
    background-color: #fac43b;
    color: #000000;
    border-radius: 5px;
    border-style: none;
    font-family: "Karla", sans-serif;
    font-style: normal;
    font-weight: 700;
    font-size: 20px;
    line-height: 28px;
    font-size: 18px;
    opacity: 100%;
    max-width: 100%;
    padding: 11px 15px 10px;
    box-shadow: 5px 5px 10px -3px rgba(0, 0, 0, 0.199),
      -5px -5px 15px 3px rgba(255, 255, 255, 0.199);
    :hover {
      opacity: 100%;
    }
  }

  .login__button:hover {
    opacity: 0.8;
  }

  .login__button:active {
    opacity: 0.4;
  }

  .modal-content {
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  .login__modal--break_text {
    margin-bottom: 40px;
  }
`;

export default LoginFormStyled;
