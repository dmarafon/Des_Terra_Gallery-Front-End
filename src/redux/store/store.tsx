import { configureStore } from "@reduxjs/toolkit";
import userReducer from "../features/userSlice";
import uiReducer from "../features/uiSlice";

const store = configureStore({
  reducer: {
    user: userReducer,
    ui: uiReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;

export default store;
