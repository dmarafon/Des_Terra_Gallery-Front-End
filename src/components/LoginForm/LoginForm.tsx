import { SyntheticEvent, useEffect, useState } from "react";
import { NavLink, useNavigate } from "react-router-dom";
import {
  apiResponseActionCreator,
  cleanApiResponseActionCreator,
} from "../../redux/features/uiSlice";
import { loginUserThunk } from "../../redux/thunks/userThunks";
import { useAppDispatch, useAppSelector } from "../hooks";
import LoadingModal from "../LoadingModal/LoadingModal";
import ModalText from "../ModalText/ModalText";
import LoginFormStyled from "./LoginFormStyled";

const LoginForm = (): JSX.Element => {
  const formInitialState = { email: "", password: "" };

  const [formData, setFormData] = useState(formInitialState);
  const navigate = useNavigate();
  const loading = useAppSelector((state) => state.ui.loading);
  const feedback = useAppSelector((state) => state.ui.feedback);
  const apiMessage = useAppSelector((state) => state.ui.apiResponse);
  const logged = useAppSelector((state) => state.user.logged);

  const dispatch = useAppDispatch();

  useEffect(() => {
    if (logged) {
      navigate("/home");
    }
  }, [logged, navigate]);

  const loginSubmit = (event: SyntheticEvent) => {
    event.preventDefault();

    switch (true) {
      case formData.email === "" && formData.password === "":
        dispatch(apiResponseActionCreator("Blank"));
        break;
      case formData.email === "":
        dispatch(apiResponseActionCreator("Email Blank"));
        break;
      case formData.password === "":
        dispatch(apiResponseActionCreator("Password Blank"));
        break;
      case formData.password.length < 5:
        dispatch(apiResponseActionCreator("Password Length"));
        break;
      default:
        const dispatchedData = { ...formData };
        resetForm();

        dispatch(loginUserThunk(dispatchedData));
    }
  };

  const submitClosingModalResponse = () => {
    dispatch(cleanApiResponseActionCreator());
  };

  const changeData = (event: SyntheticEvent) => {
    setFormData({
      ...formData,
      [(event.target as HTMLInputElement).id]: (
        event.target as HTMLInputElement
      ).value,
    });
  };

  const resetForm = () => {
    setFormData(formInitialState);
  };

  return (
    <>
      {apiMessage === "Unkwnown Error" && (
        <ModalText
          handleClose={submitClosingModalResponse}
          isOpen={false}
          customFunction={""}
        >
          Oops... We're sorry, something went wrong with our servers, try again
          later
          <p className="login__modal--break_text"></p>
        </ModalText>
      )}
      {apiMessage === "Blank" && (
        <ModalText
          handleClose={submitClosingModalResponse}
          isOpen={feedback}
          customFunction={""}
        >
          You left the Email and
          <p className="login__modal--break_text">Password field in blank</p>
        </ModalText>
      )}
      {apiMessage === "Email Blank" && (
        <ModalText
          handleClose={submitClosingModalResponse}
          isOpen={feedback}
          customFunction={""}
        >
          You left the Email field in blank.
          <p className="login__modal--break_text">Please, fill your Email.</p>
        </ModalText>
      )}
      {apiMessage === "Password Blank" && (
        <ModalText
          handleClose={submitClosingModalResponse}
          isOpen={feedback}
          customFunction={""}
        >
          <div className="login__modal--container">
            <p className="login__modal--break_text">
              You left the Password field in blank.
            </p>
            <p className="login__modal--break_text">
              Please, fill your Password.
            </p>
          </div>
        </ModalText>
      )}
      {apiMessage === "Password Length" && (
        <ModalText
          handleClose={submitClosingModalResponse}
          isOpen={feedback}
          customFunction={""}
        >
          <div className="login__modal--container">
            <p className="login__modal--break_text">
              Your Password should be longer than 5 characters
            </p>
          </div>
        </ModalText>
      )}
      {apiMessage === "Bad Request" && (
        <ModalText
          handleClose={submitClosingModalResponse}
          isOpen={feedback}
          customFunction={""}
        >
          Wrong Email or Password
          <p className="login__modal--break_text">
            {" "}
            Please, try again to Sign In
          </p>
        </ModalText>
      )}
      {loading ? (
        <LoadingModal />
      ) : (
        <LoginFormStyled>
          <div className="login__image--container">
            <div className="login__image--image_container">
              <img
                className="login__image"
                src="https://firebasestorage.googleapis.com/v0/b/desterra-181ac.appspot.com/o/leaving_the_bed.jpg?alt=media&token=1e89bee0-4d40-412c-863b-fc48a7be959a"
                alt="artist painting"
              />
            </div>
            <div className="login__image--text_container">
              <p className="login__image--subtitle">
                Nick Alm, "Leaving the Bed"
              </p>
            </div>
          </div>
          <div className="login__container">
            <p className="login__text--intro">
              SIGN IN TO{" "}
              <span className="login__text--colored">EXPERIENCE </span>
            </p>
            <p className="login__text--intro_second">DES TERRA</p>
            <form onSubmit={loginSubmit} noValidate autoComplete="off">
              <div className="login__form">
                <label className="login__label" htmlFor="email">
                  EMAIL
                </label>
                <input
                  className="login__input"
                  id="email"
                  value={formData.email}
                  onChange={changeData}
                  required
                  placeholder="EMAIL"
                  maxLength={33}
                />
                <label className="login__label" htmlFor="password">
                  PASSWORD
                </label>
                <input
                  autoComplete="current-password"
                  className="login__input"
                  id="password"
                  type="password"
                  value={formData.password}
                  onChange={changeData}
                  required
                  placeholder="PASSWORD"
                  maxLength={15}
                  minLength={5}
                />
              </div>
              <div className="login__button--container">
                <button
                  className="login__button"
                  type="submit"
                  disabled={false}
                >
                  SIGN IN
                </button>
              </div>
            </form>
            <div className="login__text--register">
              <div>
                <NavLink
                  to="/users/register"
                  className="login__register--navigation"
                >
                  Donâ€™t have an account?
                  <span className="login__text--colored"> Join Now!</span>
                </NavLink>
              </div>
            </div>
          </div>
        </LoginFormStyled>
      )}
    </>
  );
};

export default LoginForm;
