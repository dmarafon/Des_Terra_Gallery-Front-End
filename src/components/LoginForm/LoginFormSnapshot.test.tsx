import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import TestRenderer from "react-test-renderer";
import store from "../../redux/store/store";
import LoginForm from "./LoginForm";

describe("Given an Login form component", () => {
  describe("When it's rendered", () => {
    test("Then it should always match this snapshot", () => {
      const renderedAddEditForm = TestRenderer.create(
        <BrowserRouter>
          <Provider store={store}>
            <LoginForm />
          </Provider>
        </BrowserRouter>
      );

      expect(renderedAddEditForm).toMatchSnapshot();
    });
  });
});
