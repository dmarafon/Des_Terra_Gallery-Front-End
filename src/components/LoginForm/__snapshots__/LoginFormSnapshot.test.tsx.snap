// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an Login form component When it's rendered Then it should always match this snapshot 1`] = `
<div
  className="sc-dkzDqf cwWVvp"
>
  <div
    className="login__image--container"
  >
    <div
      className="login__image--image_container"
    >
      <img
        alt="artist painting"
        className="login__image"
        src="https://firebasestorage.googleapis.com/v0/b/desterra-181ac.appspot.com/o/leaving_the_bed.jpg?alt=media&token=1e89bee0-4d40-412c-863b-fc48a7be959a"
      />
    </div>
    <div
      className="login__image--text_container"
    >
      <p
        className="login__image--subtitle"
      >
        Nick Alm, "Leaving the Bed"
      </p>
    </div>
  </div>
  <div
    className="login__container"
  >
    <p
      className="login__text--intro"
    >
      SIGN IN TO
       
      <span
        className="login__text--colored"
      >
        EXPERIENCE 
      </span>
    </p>
    <p
      className="login__text--intro_second"
    >
      DES TERRA
    </p>
    <form
      autoComplete="off"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="login__form"
      >
        <label
          className="login__label"
          htmlFor="email"
        >
          EMAIL
        </label>
        <input
          className="login__input"
          id="email"
          maxLength={33}
          onChange={[Function]}
          placeholder="EMAIL"
          required={true}
          value=""
        />
        <label
          className="login__label"
          htmlFor="password"
        >
          PASSWORD
        </label>
        <input
          autoComplete="current-password"
          className="login__input"
          id="password"
          maxLength={15}
          minLength={5}
          onChange={[Function]}
          placeholder="PASSWORD"
          required={true}
          type="password"
          value=""
        />
      </div>
      <div
        className="login__button--container"
      >
        <button
          className="login__button"
          disabled={false}
          type="submit"
        >
          SIGN IN
        </button>
      </div>
    </form>
    <div
      className="login__text--register"
    >
      <div>
        <a
          className="login__register--navigation"
          href="/users/register"
          onClick={[Function]}
        >
          Donâ€™t have an account?
          <span
            className="login__text--colored"
          >
            Join Now!
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;

